packages/client/src/views/VerifyCertificate/VerifyCertificatePage.tsx

```diff
@@ -317,38 +325,40 @@ export function VerifyCertificatePage() {
     return formatDate(new Date(data.createdAt), 'dd MMMM yyyy')
   }
 
-  const getFullName = (data: any) => {
-    if (data.registration?.type === RegistrationType.Birth) {
+  const getFullName = (data: BirthRegistration | DeathRegistration) => {
+    if (isBirthRegistration(data)) {
       return (
-        data?.child?.name[0]?.firstNames + ' ' + data.child.name[0].familyName
+        data.child?.name?.[0]?.firstNames +
+        ' ' +
+        data.child?.name?.[0]?.familyName
       )
     }
+
     if (data.registration?.type === RegistrationType.Death) {
       return (
-        data?.deceased?.name[0]?.firstNames +
+        data.deceased?.name?.[0]?.firstNames +
         ' ' +
-        data?.deceased?.name[0]?.familyName
+        data.deceased?.name?.[0]?.familyName
```

Comment:
@Zangetsu101 hardcoded name generation... would probably be smart to refactor at the same time

Reply:
We should be able to use this: https://github.com/opencrvs/opencrvs-core/blob/624f6b8ac310472d3966ade57bb9131b4632d6ef/packages/client/src/utils/data-formatting.ts#L26

Reply:
okay sounds good